channels/chan_simpleusb.c

```diff
@@ -2090,6 +2104,13 @@ static struct ast_frame *simpleusb_read(struct ast_channel *c)
 				if (f1->datalen - src >= l) {	
 					/* enough to fill a frame */
 					memcpy(o->simpleusb_write_buf + o->simpleusb_write_dst, (char *) f1->data.ptr + src, l);
+
+					/* TBR - below appears to be an attempt to match levels to the original CM108
+					 * IC which has been out of production for over 10 years. Scaling audio to 
+					 * 109.375% will result in clipping! Any adjustments for CM1xxx gain differences
+					 * should be made in the mixer settings, not in the audio stream itself.
+					 */
+#if 0
```

Comment:
I very much appreciate the thoughts/comments but I question commenting out this block of code.  Won't this impact everyone with a non-C108_PRODUCT_ID device requiring them to make adjustments to their [mixer] settings?  If this is true, we will likely end up with lots of nodes with low audio.  If we do make this change now then we will need a very **BIG** release note.

Reply:
This is going to break most asl3 nodes and my nodes.  This might be appropriate on new installs we some sort of legacy option.

I would have to revisit all my sites after this update.

I think this should be in another pull request.  It is going to hold uo these changes.

Reply:
Maybe we need an 'audio_scaling' scaling variable that enables/disables this behavior.  We set the default value to be TRUE/yes (to protect existing configurations that don't have the variable in their .conf files).  Then, we add 'audio_scaling = no' to the template "simpleusb.conf" file that will be used for "new" installations.

Reply:
These sections that I if 0'd out are a major, major issue and bug, as they **will introduce distortion** in Tx audio on all ASL nodes that use any URI made in the past 10 years. I cannot stress enough how serious this is.

If incoming IAX audio happens to be less than 90%, which is usually the case for **most** users, you don't get distortion because these blocks of code are scaling by ~110%, which is only a  0.83dB level difference, but any IAX audio between 0dBFS and -0.83dBFS **will be digitally clipped and distorted** ie. sine waves squared off at the top and bottom (which has a very negative impact on audio quality and is very noticeable) on any and all nodes using ASL that have any URI which is not using the original 10+ year-old CM108 IC.

Once this digital clipping is introduced, it doesn't matter what any of your mixer settings are -- the Tx audio **will be** distorted.  Let me know if you have any questions on this or if anything I have explained above is not clear.

Fixing this now will greatly improve audio quality while bringing Tx audio levels to exactly where they should be, which is difference of -0.83dB. Differences of less than 1dB are usually not even perceptible thus fixing this will **not** cause "low" audio.

I will open an bug report on this now.

Reply:
Bug report opened: https://github.com/AllStarLink/app_rpt/issues/399

Reply:
Also see this post for additional notes: https://community.allstarlink.org/t/asl-usb-audio-notes/21832/

Reply:
> This is going to break most asl3 nodes and my nodes. This might be appropriate on new installs we some sort of legacy option.
> I would have to revisit all my sites after this update.

Hi Danny, just wanted to clarify, once this issue is fixed there will be essentially zero perceptible difference in tx audio levels. The channel drivers are now multiplying all outgoing audio by 1.1, which results in only a 0.83dB difference in audio levels - and which also results in 0.83 dB of clipping and distortion added to any incoming IAX audio signals that have peaks above -0.83dBFS.  Thus fixing this will result in less than 1dB of difference in audio levels, which with most repeaters having 50-60dB of dynamic range, is essentially completely unnoticeable. And your repeaters would then all have almost 1 dB **less** distortion, which **will** make a noticeable improvement difference in audio quality. 

Thus to be as clear, fixing this will absolutely not break anything, and you would not have to revisit any of your sites. Please see the bug report I opened for more discussion and notes on this.

The only thing that is broken is the current implementation, which is creating distortion, as well as added quantization noise. These hacks are absolutely, unquestionably inappropriate, benefit no one, and should have never been in ASL. Fortunately they are extremely easy to remove, and will result only in an extremely slight reduction in Tx audio volume, while significantly improving audio quality for all ASL users.

Reply:
PR has been updated to re-enable the audio stream scaling hacks in chan_simplusb.c and chan_usbradio.c. Moved the call to check_rx_audio() to above where the 0.8x rx audio scaling is done in chan_usbradio.c.